File: src/main/java/com/fish/entity/pojo/Notion.java
package com.fish.entity.pojo;


import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.KeyType;
import com.mybatisflex.annotation.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 *  实体类。
 *
 * @author fish
 * @since 2023-11-14
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(value = "oj_notion")
public class Notion implements Serializable {

    @Id(keyType = KeyType.Auto)
    private Long notionId;

    private String hostId;

    private String title;

    private String content;

    private Boolean enabled;

    private LocalDateTime createTime;

}

File: src/main/kotlin/com/fish/entity/pojo/Notion.kt
package com.fish.entity.pojo

import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer
import com.mybatisflex.annotation.Id
import com.mybatisflex.annotation.KeyType
import com.mybatisflex.annotation.Table
import java.io.Serializable
import java.time.LocalDateTime

/**
 * 实体类。
 *
 * @author fish
 * @since 2023-11-14
 */
@Table(value = "oj_notion")
open class Notion(
    notionId: Long?,
    hostId: String?,
    title: String?,
    content: String?,
    enabled: Boolean?,
    createTime: LocalDateTime?
): Serializable {
    @Id(keyType = KeyType.Auto)
    val notionId: Long? = null
    val hostId: String? = null
    val title: String? = null
    val content: String? = null
    val enabled: Boolean? = null
    @JsonSerialize(using = LocalDateTimeSerializer::class)
    @JsonDeserialize(using = LocalDateTimeDeserializer::class)
    val createTime: LocalDateTime? = null
}
